### PR TITLE
[tests] CA-less testcase fixes

### PR DESCRIPTION
**Pull request by Akasurde**
* Do not copy filenames from negative testcases such as 'does-not-exist'
* Add interactive 'yes'

Fixes: https://pagure.io/freeipa/issue/7182

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

### CODE REVIEW

**Comment by Akasurde** [comment]
```
# IPATEST_YAML_CONFIG=/root/mh_cfg.yml ipa-run-tests -v -r a --with-xunit test_integration/test_caless.py -k test_nonexistent_ds_pkcs12_file --junit-xml=test_nonexistent_ds_pkcs12_file.xml
===================================================== test session starts ======================================================
platform linux2 -- Python 2.7.5 -- py-1.4.32 -- pytest-2.7.0 -- /usr/bin/python2
rootdir: /usr/lib/python2.7/site-packages/ipatests/test_integration, inifile:
plugins: multihost, sourceorder
collected 96 items

test_integration/test_caless.py::TestServerInstall::test_nonexistent_ds_pkcs12_file PASSED
test_integration/test_caless.py::TestReplicaInstall::test_nonexistent_ds_pkcs12_file PASSED
test_integration/test_caless.py::TestCertinstall::test_nonexistent_ds_pkcs12_file PASSED

-------------- generated xml file: /usr/lib/python2.7/site-packages/ipatests/test_nonexistent_ds_pkcs12_file.xml ---------------
================================== 93 tests deselected by '-ktest_nonexistent_ds_pkcs12_file' ==================================
========================================== 3 passed, 93 deselected in 754.78 seconds ===========================================
```

**Comment by Akasurde** [comment]
Before ::
Testcase failed

After::
```
# IPATEST_YAML_CONFIG=/root/mh_cfg.yml ipa-run-tests -v -r a --with-xunit test_integration/test_caless.py -k test_nonexistent_http_pkcs12_file --junit-xml=test_nonexistent_http_pkcs12_file.xml
===================================================== test session starts ======================================================
platform linux2 -- Python 2.7.5 -- py-1.4.32 -- pytest-2.7.0 -- /usr/bin/python2
rootdir: /usr/lib/python2.7/site-packages/ipatests/test_integration, inifile:
plugins: multihost, sourceorder
collected 96 items

test_integration/test_caless.py::TestServerInstall::test_nonexistent_http_pkcs12_file PASSED
test_integration/test_caless.py::TestReplicaInstall::test_nonexistent_http_pkcs12_file PASSED
test_integration/test_caless.py::TestCertinstall::test_nonexistent_http_pkcs12_file PASSED

------------- generated xml file: /usr/lib/python2.7/site-packages/ipatests/test_nonexistent_http_pkcs12_file.xml --------------
================================= 93 tests deselected by '-ktest_nonexistent_http_pkcs12_file' =================================
========================================== 3 passed, 93 deselected in 739.81 seconds ===========================================
```

File: ipatests/test_integration/test_caless.py [hunk 1]
```diff
@@ -266,6 +266,10 @@ def prepare_replica(self, _replica_number=0, replica=None, master=None,
         tasks.prepare_host(master)
         tasks.prepare_host(replica)
         for filename in set(files_to_copy):
+            if filename == 'does_not_exist':
```
Comment (MartinBasti) [inline ipatests/test_integration/test_caless.py]: Hello,

I think that this is quite unsafe to hardcode such name.

Wouldn't be better to use:

```python
filepath = os.path.join(self.cert_dir, filename)
if not os.path.exists(filepath):
    logger.info("File '%s' not found, copying to server skipped", filepath)
    continue
```

Comment (Akasurde) [reply ipatests/test_integration/test_caless.py]: Sounds OK to me.

**Comment by Akasurde** [comment]
```
# IPATEST_YAML_CONFIG=/root/mh_cfg.yml ipa-run-tests -v -r a --with-xunit test_integration/test_caless.py -k test_interactive_missing_http_pkcs_password --junit-xml=test_interactive_missing_http_pkcs_password.xml
===================================================== test session starts =====================================================
platform linux2 -- Python 2.7.5 -- py-1.4.32 -- pytest-2.7.0 -- /usr/bin/python2
rootdir: /usr/lib/python2.7/site-packages/ipatests/test_integration, inifile:
plugins: multihost, sourceorder
collected 96 items

test_integration/test_caless.py::TestServerInstall::test_interactive_missing_http_pkcs_password PASSED
test_integration/test_caless.py::TestReplicaInstall::test_interactive_missing_http_pkcs_password PASSED
test_integration/test_caless.py::TestCertinstall::test_interactive_missing_http_pkcs_password PASSED

-------- generated xml file: /usr/lib/python2.7/site-packages/ipatests/test_interactive_missing_http_pkcs_password.xml --------
=========================== 93 tests deselected by '-ktest_interactive_missing_http_pkcs_password' ============================
========================================= 3 passed, 93 deselected in 1267.27 seconds ==========================================
```

```
# IPATEST_YAML_CONFIG=/root/mh_cfg.yml ipa-run-tests -v -r a --with-xunit test_integration/test_caless.py -k test_interactive_missing_ds_pkcs_password --junit-xml=test_interactive_missing_ds_pkcs_password.xml
===================================================== test session starts =====================================================
platform linux2 -- Python 2.7.5 -- py-1.4.32 -- pytest-2.7.0 -- /usr/bin/python2
rootdir: /usr/lib/python2.7/site-packages/ipatests/test_integration, inifile:
plugins: multihost, sourceorder
collected 96 items

test_integration/test_caless.py::TestServerInstall::test_interactive_missing_ds_pkcs_password PASSED
test_integration/test_caless.py::TestReplicaInstall::test_interactive_missing_ds_pkcs_password PASSED
test_integration/test_caless.py::TestCertinstall::test_interactive_missing_ds_pkcs_password PASSED

--------- generated xml file: /usr/lib/python2.7/site-packages/ipatests/test_interactive_missing_ds_pkcs_password.xml ---------
============================ 93 tests deselected by '-ktest_interactive_missing_ds_pkcs_password' =============================
========================================= 3 passed, 93 deselected in 1267.75 seconds ==========================================
```

**Comment by Akasurde** [comment]
ready_for_review

**Comment by stlaz** [comment]
It's not clear what you're trying to fix in this PR. I see failing and passing tests in some of your local runs which does not really help anything. Could you explain what you're doing here?

**Comment by Akasurde** [comment]
@stlaz I am fixing four failing testcases

1. test_nonexistent_ds_pkcs12_file
2. test_nonexistent_http_pkcs12_file 

These testcases are failing as they are trying to copy non-existent file to system under tests. So I am checking if file exists on controller node then only copy that file to system under tests.

3. test_interactive_missing_ds_pkcs_password
4. test_interactive_missing_http_pkcs_password

These testcases are failing as they are missing one interactive prompt in ipa-replica-install command. So I am adding new "yes" to that interactive prompt.

Let me know if it helps.

**Comment by Akasurde** [comment]
As per comments, closing this PR in-favour of two separate PR
- https://github.com/freeipa/freeipa/pull/1142
- https://github.com/freeipa/freeipa/pull/1143

**Pull request closed without merging**